gramps/gen/lib/json_utils.py

```diff
@@ -0,0 +1,219 @@
+            )
+
+        if "_object" in self:
+            return getattr(self["_object"], key)
+        elif key in self:
+            value = self[key]
+            if isinstance(value, dict):
+                return DataDict(value)
+            elif isinstance(value, list):
+                return DataList(value)
+            else:
+                return value
+        else:
+            self["_object"] = data_to_object(self)
+            return getattr(self["_object"], key)
+
+
+class DataList(list):
+    """
+    A wrapper around a data list.
+    """
+
+    def __getitem__(self, position):
+        value = super().__getitem__(position)
+        if isinstance(value, dict):
+            return DataDict(value)
+        elif isinstance(value, list):
+            return DataList(value)
+        else:
+            return value
+
+
+def convert_state_to_object(obj_dict):
+    _class = obj_dict.pop("_class")
+    cls = lib.__dict__[_class]
+    obj = cls.__new__(cls)
+    obj.set_object_state(obj_dict)
+    return obj
+
+
+def convert_object_to_state(obj):
+    return obj.get_object_state()
+
+
+def string_to_data(string: str | bytes) -> DataDict:
+    """
+    Convert a JSON string into its data representation.
+    """
+    return DataDict(orjson.loads(string))
+
+
+def string_to_dict(string: str | bytes) -> dict:
+    """
+    Convert a JSON string into its dict representation.
+    """
+    return orjson.loads(string)
+
+
+def dict_to_string(dict: dict) -> str | bytes:
+    """
+    Convert a dict into its JSON string representation.
+    """
+    return orjson.dumps(dict)
+
+
+def object_to_dict(obj):
+    """
+    Convert any Gramps lib object into its DataDict representation.
+    """
+    if isinstance(obj, (float, int, str, NoneType)):
+        return obj
+
+    elif isinstance(obj, (list, tuple)):
+        return [object_to_dict(item) for item in obj]
+
+    state = obj.get_object_state()
+    return {k: object_to_dict(v) for k, v in state.items()}
+
+
+def object_to_data(obj):
+    """
+    Convert any Gramps lib object or other value into
+    its dict representation or value, respectively.
+    """
+    if isinstance(obj, (float, int, str, NoneType)):
+        return obj
+
+    elif isinstance(obj, (list, tuple)):
+        return DataList([object_to_data(item) for item in obj])
+
+    state = obj.get_object_state()
+    data = {k: object_to_data(v) for k, v in state.items()}
+    # Stash the object in case we need it later:
+    data["_object"] = obj
+    return DataDict(data)
+
+
+def data_to_object(data):
+    """
+    Convert any object dict representation to a Gramps lib object
+    or other value.
+    """
+    if isinstance(data, dict):
+        if "_object" in data:
+            return data["_object"]
+
+        data = {k: data_to_object(v) for k, v in data.items()}
+        return convert_state_to_object(data)
+
+    elif isinstance(data, (list, tuple)):
```

Comment:
```suggestion
    if isinstance(data, (list, tuple)):
```
Else after return.

Reply:
Performance is the same. "Explicit is better than implicit."